### PR TITLE
objectcreator: linux makefile

### PR DESCRIPTION
**Pull request by andrei8l**
#### Summary

SUMMARY: Build "Rudimentary makefile for ObjectCreator tool"

#### Purpose of change

Add a Makefile so we can compile ObjectCreator (https://github.com/CleverRaven/Cataclysm-DDA/pull/43171) on linux

#### Describe the solution
Bang head

#### Describe alternatives you've considered
meson
<details>
<summary>snippet from my meson build definition</summary>

```
objectcreator = executable(
  'object_creator',
  [
    'object_creator/creator_main.cpp',
    'object_creator/creator_main_window.cpp',
    'object_creator/dual_list_box.cpp',
    'object_creator/fake_spell_listbox.cpp',
    'object_creator/fake_spell_window.cpp',
    'object_creator/moc_dual_list_box.cpp',
    'object_creator/moc_fake_spell_listbox.cpp',
    'object_creator/moc_fake_spell_window.cpp',
    'object_creator/spell_window.cpp',
    'src/messages.cpp',
    'tools/format/format.cpp'
  ],
  objects : cddatarget.extract_objects(lib_sources),
  include_directories : [ 'tools/format', 'src' ],
  install : false,
  dependencies : [
    deps,
    dependency('qt5', modules: ['Core', 'Gui', 'Widgets']) 
  ],
  build_by_default : false,
)
```
</details>

#### Testing
`make object_creator`

#### Additional context
@KorGgenT please test that it still works on windows. I had to make a minor change wrt plugin import

### CODE REVIEW

File: object_creator/fake_spell_window.h [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #ifndef CATA_OBJECT_CREATOR_FAKE_SPELL_WINDOW_H
 #define CATA_OBJECT_CREATOR_FAKE_SPELL_WINDOW_H
 
+#undef emit
```
Comment (ZhilkinSerg) [inline object_creator/fake_spell_window.h:4]: Isn't it handled by `QT_NO_KEYWORDS` define?

Comment (andrei8l) [reply object_creator/fake_spell_window.h]: Didn't know about it. Thanks!

**Comment by KorGgenT** [comment]
no errors in vs

**Comment by ZhilkinSerg** [comment]
Builds with MSYS too.

Install `qt5`:

```
pacman -S mingw-w64-x86_64-qt5
```

Then build with something like:

```
make object_creator CCACHE=1 MSYS2=1 BACKTRACE=1 DYNAMIC_LINKING=1 RELEASE=1 LINTJSON=0 TILES=1 SDL=1 SOUND=1 ASTYLE=0 LOCALIZE=1 RUNTESTS=1 -j8 LANGUAGES=all
```

**Comment by andrei8l** [comment]
Nice; thanks for checking!

**Pull request merged**